src/pages/index.js

```diff
@@ -23,45 +34,172 @@ import {
   const userProfileData = userProfile.getUserInfo();
   editPopup.setInputValues(userProfileData);
 }
 
+function handleAvatarSubmit(inputObj) {
+  avatarEditPopup.showUploadingState();
+  const avatarLink = inputObj[avatarLinkInputId];
+  api
+    .updateProfileImage(avatarLink)
+    .then(avatarEditPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  avatarImg.src = avatarLink;
+  avatarEditPopup.close();
+  formValidators[avatarFormId].disableSubmitButton();
+  avatarEditPopup.reset();
+}
+
 function handleProfileSubmit(inputObj) {
-  userProfile.setUserInfo(
-    inputObj[editInputTitleId],
-    inputObj[editInputDescriptionId]
-  );
+  const nameEl = inputObj[editInputTitleId];
+  const jobEl = inputObj[editInputDescriptionId];
+  editPopup.showUploadingState();
+  api
+    .updateUserInfo({
+      editFormNameInput: nameEl,
+      editFormAboutInput: jobEl,
+    })
+    .then(editPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  // sent inputs to server
+  userProfile.setUserInfo(nameEl, jobEl);
+  // get userinfo from server and apply to profile
   editPopup.close();
+  formValidators[profileEditFormId].disableSubmitButton();
+  editPopup.reset();
 }
 function creatCard(item) {
-  const cardElement = new Card(item, cardTempSelector, handleImageClick);
+  const cardElement = new Card(
+    item,
+    cardTempSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardItem = cardElement.getView();
   return cardItem;
 }
+
 function handleAddCardSubmit(inputObj) {
   addCardPopup.close();
-  const newCard = creatCard({
-    name: inputObj[addCardInputTitleId],
-    link: inputObj[addCardInputLinkId],
-  });
-  cardList.addItem(newCard);
-  formValidators[addCardFormId].disableSubmitButton();
-  addCardPopup.reset();
+  const cardName = inputObj[addCardInputTitleId];
+  const cardLink = inputObj[addCardInputLinkId];
+  addCardPopup.showUploadingState();
+  api
+    .addNewCard({ cardElementName: cardName, cardElementLink: cardLink })
+    .then((card) => {
+      addCardPopup.showUploaded();
+      const newCard = creatCard(card);
+      cardList.addItem(newCard);
+      formValidators[addCardFormId].disableSubmitButton();
+      addCardPopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 function handleImageClick({ name, link }) {
   imageInfo.open({ name, link });
 }
+function handleLikeClick(card) {
+  const actionPromise = card.isLiked
+    ? api.unlikeCard(card).catch((err) => {
+        console.error(err);
+      })
+    : api.likeCard(card).catch((err) => {
+        console.error(err);
+      });
+  actionPromise
+    .then((updatedCard) => {
+      card.setLikeAction();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    card.deleteCard();
```

Comment:
Needs correction: 

1) please delete the card only after the successful server response
2) notify the user about loading by changing the button text to `Loading..`

Reply:
Still needs correction